DotNetCoreMasters/DotNetCoreMasters/Filter/EnsureItemExistAttribute.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Services.Interface;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace DotNetCoreMasters.Filter
+{
+    public class EnsureItemExistsFilter : IActionFilter
+    {
+        private readonly IItemService _service;
+
+        public EnsureItemExistsFilter(IItemService service)
+        {
+            _service = service;
+        }
+        public void OnActionExecuted(ActionExecutedContext context) { }
+
+        public void OnActionExecuting(ActionExecutingContext context)
+        {
+            var itemId = (int)context.ActionArguments["itemId"];
```

Comment:
Sir we need to handle scenarios where the argument doesn't exist because we are applying this filter on the controller scope